### PR TITLE
Interruptible instances scaling , mostly by using existing clusteraut…

### PR DESCRIPTION
**Pull request by miyadav**
…oscaler tests
@sunzhaohua2  , PTAL , if we shall proceed with this ? , trying to reuse the clusterautoscaler test written by @jhou1  and interruptible instance test by you ..to cover this scenario for autoruns

### CODE REVIEW

File: features/machine/machine.feature [hunk 1]
```diff
@@ -202,6 +202,59 @@ Feature: Machine features testing
...
+      | f | cluster-autoscaler.yml |
+    Then the step should succeed
+    And admin ensures "default" clusterautoscaler is deleted after scenario
+    And 1 pod becomes ready with labels:
+      | cluster-autoscaler=default,k8s-app=cluster-autoscaler |
+
+    # Create machineautoscaler
+    Given I obtain test data file "cloud/machine-autoscaler.yml"
+    When I run oc create over "machine-autoscaler.yml" replacing paths:
+      | ["metadata"]["name"]               | maotest                 |
```
Comment (sunzhaohua2) [inline features/machine/machine.feature:217]:
```
      | ["metadata"]["name"]               | maotest           |
      | ["spec"]["minReplicas"]            | 1                 |
      | ["spec"]["maxReplicas"]            | 3                 |
      | ["spec"]["scaleTargetRef"]["name"] | <machineset_name> |
```

**Comment by sunzhaohua2** [comment]
If the verification is passed, this is LGTM.

**Comment by miyadav** [comment]
> If the verification is passed, this is LGTM.

Here are results :


Here is the functional validation , https://privatebin-it-iso.int.open.paas.redhat.com/?071e3f61d4e66c59#4ixWBAwrGyL2fb4khtWWZdTs8h7fTm6w69dVt1MJS6eN

Checked for AWS spot instances looks good 
```
.
.
    kind: List
    metadata:
      resourceVersion: ""
      selfLink: ""
    
    [06:16:52] INFO> Exit Status: 0
    [06:16:52] INFO> Shell Commands: oc delete machines machineset-clone-29199-tcmdg --wait=false --config=/root/workdir/79aafb0e5ead-/ocp4_admin.kubeconfig --namespace=openshift-machine-api
    machine.machine.openshift.io "machineset-clone-29199-tcmdg" deleted
    
    [06:16:54] INFO> Exit Status: 0
    [06:16:55] INFO> oc get machines machineset-clone-29199-tcmdg --output=yaml --config=/root/workdir/79aafb0e5ead-/ocp4_admin.kubeconfig --namespace=openshift-machine-api
    [06:17:20] INFO> After 12 iterations and 26 seconds:
    
    STDERR:
    Error from server (NotFound): machines.machine.openshift.io "machineset-clone-29199-tcmdg" not found
    
    [06:17:20] INFO> Shell Commands: oc get machines machineset-clone-29199-tcmdg --output=yaml --config=/root/workdir/79aafb0e5ead-/ocp4_admin.kubeconfig --namespace=openshift-machine-api
    
    STDERR:
    Error from server (NotFound): machines.machine.openshift.io "machineset-clone-29199-tcmdg" not found
    
    [06:17:22] INFO> Exit Status: 1
    [06:17:22] INFO> Shell Commands: rm -r -f -- /root/workdir/79aafb0e5ead-
    
    [06:17:23] INFO> Exit Status: 0
    [06:17:24] INFO> === End After Scenario: Required configuration should be added to the ProviderSpec to enable spot instances, Examples (#1) ===

1 scenario (1 passed)
38 steps (38 passed)
10m24.142s
[06:17:24] INFO> === At Exit ===
```

File: features/machine/machine.feature (hunk 1 above)
Comment (miyadav) [reply features/machine/machine.feature]: resolved and squashed commit

**Pull request merged**